src/devices/bus/snes/upd.cpp

```diff
@@ -142,14 +142,18 @@ void sns_rom20_necdsp_device::device_add_mconfig(machine_config &config)
 uint8_t sns_rom20_necdsp_device::chip_read(offs_t offset)
 {
 	offset &= 0x7fff;
-	return m_upd7725->snesdsp_read(offset < 0x4000);
+	if (BIT(offset, 14))
+		return m_upd7725->status_r();
+	else
+		return m_upd7725->data_r();
 }
 
 
 void sns_rom20_necdsp_device::chip_write(offs_t offset, uint8_t data)
 {
 	offset &= 0x7fff;
-	m_upd7725->snesdsp_write(offset < 0x4000, data);
+	if (BIT(~offset, 14))
+		m_upd7725->data_w(data);
 }
```

Comment:
Same here, you can get rid of the `offset &= 0x7fff`, and I’d like someone to double-check whether bit 14 really does stop it from writing.